CMakeLists.txt

```diff
@@ -106,39 +106,9 @@ if(SFML_BUILD_GRAPHICS AND NOT SFML_BUILD_WINDOW)
     set(SFML_BUILD_WINDOW TRUE)
 endif()
 
-# allow not using bundled dependencies with a switch
-# (except for stb_image)
-# yes this is horrible, but GLOB_RECURSE sucks
 sfml_set_option(SFML_USE_SYSTEM_DEPS FALSE BOOL "TRUE to use system dependencies, FALSE to use the bundled ones.")
```

Comment:
Maybe with this change it would be better to default this to `TRUE` (maybe only on platforms other than Windows)? Or it could essentially be a tri-state between "Only use system dependencies", "Build all dependencies" or "Build only missing dependencies"?

Reply:
I think we have to look at it from a user experience perspective.

Experienced users are going to know this option exists and know whether they want to build using system dependencies or not and what that means in terms of requirements e.g. installing development libraries, having to install packages on target systems as well etc.

Beginner users who just wanted to get access to the master version of SFML might not know what using system dependencies means and if it is even viable on the system they are developing on/for.

I think leaving the default at a value that is going to work better for beginner users is the better choice.

Reply:
Didn't it used system libraries on Linux before this PR? (e.g. [src/SFML/Graphics/CMakeLists.txt#L101-L110](https://github.com/SFML/SFML/blob/master/src/SFML/Graphics/CMakeLists.txt#L101-L110) didn't set the include directory on Linux)

I haven't taken a proper look at this PR yet, but I imagine that you would then also need to install all the dependencies such as freetype to /usr/local/ in order for the application to use SFML? Couldn't it cause issues for other projects that there is suddenly a freetype include and lib available in /usr/local that overrules the system one in /usr? (On Windows that would be less of a problem because you install to some custom folder that doesn't affect the rest of the system).

If you already considered the consequences of switching to bundled libraries on linux and you find that there shouldn't be any problems with it with either static or dynamic linking then I guess it's probably fine to use bundled libs by default.

Reply:
I guess SFML did always use system dependencies on Linux. Now that I see it I don't think it is very intuitive to provide `SFML_USE_SYSTEM_DEPS` and for it not to have an effect when building on Linux. Since we can't make any assumptions on systems we haven't yet encountered, the default for `SFML_USE_SYSTEM_DEPS` should still be `FALSE` but it would make sense to set the default to `TRUE` on Linux to mimic the old behaviour.

>  haven't taken a proper look at this PR yet, but I imagine that you would then also need to install all the dependencies such as freetype to /usr/local/ in order for the application to use SFML? Couldn't it cause issues for other projects that there is suddenly a freetype include and lib available in /usr/local that overrules the system one in /usr? (On Windows that would be less of a problem because you install to some custom folder that doesn't affect the rest of the system).

/usr/local should never be touched by the system package manager. It is the place that "local system configuration" i.e. self-built libraries and configuration files should be placed into. It is up to the user to maintain it. I've already built many things that ended up being installed to /usr/local by default. Because the search paths are set to prioritize /usr/local it will always be preferred over anything else if the user doesn't explicitly specify anything.

Reply:
I guess since we still require some OS dependencies (X11, udev, etc.) on Linux, it should be fine to also recommend installing the system version for our dependencies.

With `brew` being the defacto package manager on macOS, it's not official and doesn't get shipped by default, so we can't make the same assumption.  
Windows is actually further ahead with `winget` being officially supported by Microsoft and coming pre-installed (on Windows 11?), but then again Windows doesn't really have a (useful) way to install development libraries and also doesn't seem to have pure libraries.